etcdctlv3/command/put_command.go

```diff
@@ -31,9 +31,31 @@ var (
 // NewPutCommand returns the cobra command for "put".
 func NewPutCommand() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "put",
+		Use:   "put [options] <key> <value>",
 		Short: "Put puts the given key into the store.",
-		Run:   putCommandFunc,
+		Long: `
+When <value> has first character of '-', it is interpreted as a flag.
+Insert '--' for workaround:
+
+$ put [options] <key> -- <value>
+$ put [options] -- <key> <value>
+
+NOTES:
+
+This will error:
```

Comment:
OK will remove the notes part. Online usage should be enough.

On Tue, Jan 12, 2016, 17:51 Xiang Li <notifications@github.com> wrote:

> In etcdctlv3/command/put_command.go
> <https://github.com/coreos/etcd/pull/4193#discussion_r49542906>:
>
> >  		Short: "Put puts the given key into the store.",
> > -		Run:   putCommandFunc,
> > +		Long: `
> > +When <value> has first character of '-', it is interpreted as a flag.
> > +Insert '--' for workaround:
> > +
> > +$ put [options] <key> -- <value>
> > +$ put [options] -- <key> <value>
> > +
> > +NOTES:
> > +
> > +This will error:
>
> @gyuho <https://github.com/gyuho> @heyitsanthony
> <https://github.com/heyitsanthony> @jonboulle
> <https://github.com/jonboulle> Personally, I feel this example is too
> long. I feel the help message is good enough.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/coreos/etcd/pull/4193/files#r49542906>.
>
-- 
Sincerely,
Gyu-Ho Lee